moved/src/move_execution/evm_native/native_evm_context.rs

```diff
@@ -93,21 +122,18 @@ impl<'a> DatabaseRef for ResolverBackedDB<'a> {
     }
 
     fn storage_ref(&self, address: Address, index: U256) -> Result<U256, Self::Error> {
-        let struct_tag = account_storage_struct_tag(&address, &index);
-        let value = match self
-            .resolver
-            .get_resource(&EVM_NATIVE_ADDRESS, &struct_tag)?
-        {
-            Some(bytes) => {
-                let value = Value::simple_deserialize(&bytes, &ACCOUNT_STORAGE_LAYOUT)
-                    .expect("EVM account storage must deserialize correctly");
-                value.value_as::<u256::U256>()?.to_u256()
-            }
-            None => {
-                // Zero is the default value when there is no entry
-                return Ok(U256::ZERO);
-            }
-        };
+        let mut cache_lock = self
+            .storage_cache
+            .write()
```

Comment:
If we only use write lock, then we don't benefit from using an RwLock.

Although I don't think we are going to have multiple readers in any case.

Reply:
As discussed on the call, the `RwLock` is primarily giving us interior mutability for the cache as opposed to multi-threaded access. I have a comment describing the purposes of the the cache and lock in https://github.com/MovedNetwork/op-move/pull/190/commits/2e778d0043f7fb235e9deb88d73b7070a8afe675